### ISSUE TITLE
Model.predict output not consistent for different batch_sizes

### ISSUE THREAD

**Issue by jmgo** [keras-team/keras#12557]
I have created a simple CNN and tested it in dummy data. 

The outputs are different depending on the batch_size used in model.predict, except if I lower the input the be lower than (41, 41, 41). 

For instance, if the input is (40, 40, 40), there are no differences (if the print statements values are 0 then there are no differences).   

I am using keras 2.2.4, cuda 8, cudnn 6.0.21 and tensorflow-gpu =1.4.0. 

I have tried running in the CPU (by using os.environ['CUDA_VISIBLE_DEVICES'] = '-1' before importing keras), Theano backend and keras 1.1.12, but they same behaviour was observed.

Does anyone else experience the same problem by running the code below? 

```
import numpy as np
from numpy.random import randint
from keras.layers import Conv3D, Input, BatchNormalization, Softmax
from keras.models import Model

#Create model variable ###################
imgD, imgH, imgW, n_cls = (41,41,41,2)

input_layer = Input((imgD, imgH, imgW, 1), name='data')
conv_1      = Conv3D(32   , (5, 5, 5), padding='same')(input_layer)
bn_1        = BatchNormalization()(conv_1)
conv_2      = Conv3D(n_cls, (1, 1, 1), padding='same')(bn_1)
softmax     = Softmax(axis=-1)(conv_2)

model = Model(input_layer, softmax)
model.summary(line_length=113)
# model.compile(optimizer='sgd', loss='categorical_crossentropy')
##########################################

# Create random inputs ###################
n_samples = 100
imgs_train = np.float32(randint(0, 255, (n_samples, imgD, imgH, imgW, 1))) / 255
##########################################

# predict for different batch sizes ######
batchSizes  = [1,5,32,53,98] #list(range(1,40))[::-1]
msks_pred32 = model.predict(imgs_train, batch_size=32)

for i in batchSizes:
    msks_pred = model.predict(imgs_train, batch_size=i)
    print('batch size :', i, 'Diff sum (relative to batch_size = 32):', np.sum(msks_pred32 - msks_pred))
###########################################
```

Output example with input size (41, 41, 41)

```
batch size : 1 Diff sum (relative to batch_size = 32): 9.07481e-05
batch size : 5 Diff sum (relative to batch_size = 32): 9.07481e-05
batch size : 32 Diff sum (relative to batch_size = 32): 0.0
batch size : 53 Diff sum (relative to batch_size = 32): -0.000500441
batch size : 98 Diff sum (relative to batch_size = 32): 9.07481e-05
```

Output example with input size (40, 40, 40)

```
batch size : 1 Diff sum (relative to batch_size = 32): 0.0
batch size : 5 Diff sum (relative to batch_size = 32): 0.0
batch size : 32 Diff sum (relative to batch_size = 32): 0.0
batch size : 53 Diff sum (relative to batch_size = 32): 0.0
batch size : 98 Diff sum (relative to batch_size = 32): 0.0
```

**Comment by jvishnuvardhan** [comment]
@jmgo I don't see any mismatch as you are experiencing. Please take a look the output. Could you upgrade to latest version and check whether the mismatch persist? Thanks!

I am using TF1.13.1 and keras 2.2.4.

Using TensorFlow backend.
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
_________________________________________________________________________________________________________________
Layer (type)                                      Output Shape                                  Param #          
=================================================================================================================
data (InputLayer)                                 (None, 41, 41, 41, 1)                         0                
_________________________________________________________________________________________________________________
conv3d_1 (Conv3D)                                 (None, 41, 41, 41, 32)                        4032             
_________________________________________________________________________________________________________________
batch_normalization_1 (BatchNormalization)        (None, 41, 41, 41, 32)                        128              
_________________________________________________________________________________________________________________
conv3d_2 (Conv3D)                                 (None, 41, 41, 41, 2)                         66               
_________________________________________________________________________________________________________________
softmax_1 (Softmax)                               (None, 41, 41, 41, 2)                         0                
=================================================================================================================
Total params: 4,226
Trainable params: 4,162
Non-trainable params: 64
_________________________________________________________________________________________________________________
batch size : 1 Diff sum (relative to batch_size = 32): 0.0
batch size : 5 Diff sum (relative to batch_size = 32): 0.0
batch size : 32 Diff sum (relative to batch_size = 32): 0.0
batch size : 53 Diff sum (relative to batch_size = 32): 0.0
batch size : 98 Diff sum (relative to batch_size = 32): 0.0

Output example with input size (40, 40, 40)

batch size : 1 Diff sum (relative to batch_size = 32): 0.0
batch size : 5 Diff sum (relative to batch_size = 32): 0.0
batch size : 32 Diff sum (relative to batch_size = 32): 0.0
batch size : 53 Diff sum (relative to batch_size = 32): 0.0
batch size : 98 Diff sum (relative to batch_size = 32): 0.0

**Comment by jmgo** [comment]
Thanks for checking that jvishnuvardhan! 

I have not tried to use the lastest tensorflow version because it also requires to install a different version of CUDA and cuDNN. 

I will try to get the latest version of tensorflow, but it may take a while, since I am focusing on another thing right now. 

Meanwhile, could you try to run the same example with a larger input size? There may be something in you setup that may be biasing the results (e.g., GPU memory).

**Comment by jvishnuvardhan** [comment]
I am closing as it was resolved. I had used Google [colab](https://colab.sandbox.google.com/notebooks/welcome.ipynb) to run many of my models without any problem. You could do the same.  I am closing the issue. But, please let me know if I'm mistaken. Thanks!
